addons/point_of_sale/models/stock_picking.py

```diff
@@ -138,6 +138,13 @@ def _compute_hide_reservation_method(self):
             if picking_type == picking_type.warehouse_id.pos_type_id:
                 picking_type.hide_reservation_method = True
 
+    @api.constrains('active')
+    def _check_active(self):
+        for picking_type in self:
+            pos_configs = self.env['pos.config'].search([('picking_type_id', '=', picking_type.id)], limit=1)
+            if pos_configs:
+                raise ValidationError(_("You cannot archive '%s' as it is used by a POS configuration.") % picking_type.name)
```

Comment:
It would be helpful to the user to also show the pos.config records' names. Providing that info here will help the user in finding the pos.config that depends on the picking type they want to delete.

Reply:
Hello @caburj ,
Required changes are done, can you please review them?